src/MetaMorpho.sol

```diff
@@ -273,18 +268,15 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
     /// @inheritdoc IMetaMorphoBase
     function submitCap(MarketParams memory marketParams, uint256 newSupplyCap) external onlyCuratorRole {
         Id id = marketParams.id();
+        uint256 supplyCap = config[id].cap;
         if (marketParams.loanToken != asset()) revert ErrorsLib.InconsistentAsset(id);
         if (MORPHO.lastUpdate(id) == 0) revert ErrorsLib.MarketNotCreated();
-
-        uint256 supplyCap = config[id].cap;
+        if (pendingCap[id].validAt != 0) revert ErrorsLib.AlreadyPending();
         if (newSupplyCap == supplyCap) revert ErrorsLib.AlreadySet();
 
         if (newSupplyCap < supplyCap) {
             _setCap(id, newSupplyCap.toUint184());
         } else {
-            // newSupplyCap > supplyCap >= 0 so there's no need to check `pendingCap[id].validAt != 0`.
-            if (newSupplyCap == pendingCap[id].value) revert ErrorsLib.AlreadyPending();
```

Comment:
we should keep reverting at this place IMO for the reason mentioned above

Reply:
I don't get this: this check is weaker than the one introduced, why would we keep it?
Meaning that if the check introduced passes, this one should pass too

Reply:
I don't get it: if the check introduced passes (no pending value), then this one should not pass (the new value cannot equal the pending value)
Why would we keep this check?

Reply:
This enforces [this](https://github.com/morpho-org/metamorpho/pull/353#discussion_r1428051554). Which I think is very annoying... In case we still want to push that change, we must provide better explanations on the process of setting a cap to 0 in case of emergency in case a pending value has been submitted recently.

Reply:
Alright so I wouldn't argue that we should keep this check but instead move the check introduced in place of this check, so we keep the feature that bypasses the timestamp

It indeed feels weird to have the feature to bypass the timestamp but only when there's not already a pending value
But it's not that weird either: it's like having a pending value that goes in one direction but still being able to instantly push a value in the other direction

Reply:
I'm also leaning towards having a special case and allowing setting a cap to 0 even with a pending cap. In practice there is a small difference between submitting a call and submitting a bundle (and between verifying each).

Reply:
I agree that it should be documented and that setting a cap to 0 should be made easy, but I would not go as far as introducing a special case for it. Adding a different operational method for handling pending values is also a mental overhead.

As @Rubilmax said, I don't find it so weird to have to revoke a pending value if the admin wants to actually go in the other direction of the pending value